TINY-6666: Moved table cell type operations to snooker (#7057)
tinymce/tinymce: modules/snooker/CHANGELOG.md

### Added
- `Warehouse` now contains the colgroup elements from the table.
- `TableOperations` now includes functions to convert individual table cells to header cells (and vice-versa).

### Improved
- Rows are now stored and tracked in the table model structures.

### Changed
- `RowDetails` has been renamed to `RowDetail` and now takes a generic argument to define the type of cells stored.
- `row` and `colgroup` generators are now passed the previous element during transform or modification operations.

> - Renamed `getColType` to `getColsType` in the `TableOperations` module to reflect it works with multiple columns.

Author: `supports` isn't the right word here and needs to be closer to `works`, so I'm not keen on that. Happy to other suggestions though.

> - Renamed `getColType` to `getColsType` in the `TableOperations` module to reflect it works with multiple columns.

Author: ```suggestion
- Renamed `getColType` to `getColsType` in the `TableOperations` module to reflect it can lookup the type from multiple columns.
```
Maybe?

> - The `getCellsType` in the `TableOperations` module has been changed to be consistent with the `getColsType` function.

Author: It doesn't need to say exactly what it does, only that it's been changed. Plus anyone who was using this is going to know how the rest of the `TableOperations` API.
